### PR TITLE
feat(input): add KeyComboList overloads to InputManager

### PR DESCRIPTION
**Pull request by tkhquang**
## Summary

- Add `register_press` and `register_hold` overloads that accept `Config::KeyComboList` directly, eliminating boilerplate when bridging config-parsed key combos to `InputManager`
- All combos in the list share the same binding name, enabling OR-logic via `is_binding_active()`
- Add 8 new unit tests covering empty lists, single/multi-combo, shared name queries, mixed usage, and ignored-while-running behavior
- Update README code example and feature description to reflect the new API

## Release Notes

* **New Features**
  * Added new `register_press` and `register_hold` API overloads that accept multiple key combinations in a single call, simplifying binding registration without requiring per-combo iteration loops.

* **Documentation**
  * Updated configuration examples and feature documentation to demonstrate the streamlined key binding registration using the new `KeyComboList`-direct registration capability.

### CODE REVIEW

File: src/input.cpp [hunk 1]
```diff
@@ -467,6 +468,24 @@ namespace DetourModKit
...
+    }
+
+    void InputManager::register_hold(const std::string &name, const Config::KeyComboList &combos,
+                                     std::function<void(bool)> callback)
+    {
+        for (const auto &combo : combos)
+        {
+            register_hold(name, combo.keys, combo.modifiers, callback);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (tkhquang) [reply src/input.cpp:487]: The interleaving scenario requires `start()` to race with `register_press()`
from a different thread. This violates the documented API contract — registration
is a pre-start phase, and `start()` is called once after all bindings are
registered (documented in both the header and README).

More importantly, this is not a regression introduced by the batch overload.
The identical race exists for any sequence of individual `register_press()`
calls — including the manual loop pattern this overload replaces. Making only
the batch overload atomic while leaving sequential individual calls non-atomic
would create an inconsistent and misleading guarantee.

If we ever need atomic batch registration (e.g., for runtime re-binding), the
right approach is a dedicated `register_batch()` method with clear semantics,
not silently upgrading one overload's locking granularity beyond what the rest
of the API provides.
